src/arch/riscv/common_riscv.lds

```diff
@@ -43,12 +43,11 @@ SECTIONS
 
     /* Start of data section */
     _sdata = .;
-    .sdata : {
-        __global_pointer$ = . + 0x800;
-        *(.sdata*)
-    }
-    .srodata : {
+    .small : {
         *(.srodata*)
+        __global_pointer$ = .;
```

Comment:
My understanding is that if we wanted to maximize the number of symbols that can have their addressing relaxed then the `__global_pointer` symbol should be placed where the most symbols are +/- 0x800 from its definition.  Looking at `kernel_all.c.obj`, .srodata is only 0x210 in size, so it seems that something like the following gets better coverage?
```
    .small :  AT(ADDR(.small) - KERNEL_OFFSET)
    { 
        __global_pointer$ = . + 0x800;
        *(.srodata*)
        *(.sdata*)
        *(.sbss)
    }
```